pkg/ast/spl/spl.peg

```diff
@@ -2030,6 +2036,107 @@ Subseconds <- ("us" / "ms" / "cs" / "ds") {
     return timeUnit, nil
 }
 
+// Returns *structs.QueryAggregators
+TransactionBlock <- PIPE CMD_TRANSACTION txnOptions:TransactionOptions {
+    queryAgg := &structs.QueryAggregators{
+        PipeCommandType: structs.TransactionType,
+        TransactionArguments: txnOptions.(*structs.TransactionArguments),
+    }
+    return queryAgg, nil
+}
+
+// The List of fields should Always be given at the last.
+// The TransactionDefinitionOptionsList will consist of all the Transaction Options
+TransactionOptions <- txnOptions:TransactionDefinitionOptionsList? SPACE? fields:FieldList?  {
+
+    transactionRequest := &structs.TransactionArguments{
+        Fields:     nil,
+        StartsWith: "",
+        EndsWith:   "",
+    }
+
+    if fields != nil {
+        transactionRequest.Fields = fields.([]string)
+    }
+     
+    if txnOptions != nil {
+        txnArgs := txnOptions.(*TxnArgs)
+        transactionRequest.StartsWith = txnArgs.startsWith
+        transactionRequest.EndsWith = txnArgs.endsWith
+    }
+
+    return transactionRequest, nil
+}
+
+// Parses one or more FieldNames separate by a space
+// Returns a slice containing all the FieldNames
+FieldList <- first:FieldName rest:(SPACE FieldName)* {
+     var fields []string
+    fields = append(fields, first.(string))
+    for _, r := range rest.([]any) {
+        // Extracting the field name from the tuple (SPACE, FieldName)
+        fields = append(fields, r.([]any)[1].(string))
+    }
+    return fields, nil
+}
+
+
+TransactionDefinitionOptionsList <- first:TransactionDefinitionOption rest:(SPACE TransactionDefinitionOption)* {
+
+    restSlice := rest.([]any)
+    txnArgs := &TxnArgs{
+        argOption: "txn-definition",
```

Comment:
Is `"txn-definition"` just used as an invalid default? If so it would be good to rename it to something like `txn-invalid-option` so it's clearer that it is invalid. Or maybe I'm missing something?

Reply:
- The arguments `startswith` and `endswith` are classified as `txn-definition` options in Splunk. And there are more like `maxpause`, `maxspan`, etc. coming under `txn-definition` options. 

- Splunk also supports `memcontrol-options`, `rendering-options`, and `transaction-name` options with its own set of multiple optional arguments. 

- So I just made this classification for now. If we add more arguments depending on the classification on Splunk, we can also similarly group while parsing the spl command.